sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -378,14 +372,233 @@ private static ExpandoObject BuildBindingMetadataFromAttribute(CustomAttribute a
             bindingDict["Type"] = bindingType;
             bindingDict["Direction"] = GetBindingDirection(attribute);
 
+            // Is string parameter type
+            if (IsStringType(parameterType.FullName))
+            {
+                bindingDict["DataType"] = "String";
+            }
+            // Is binary parameter type
+            else if (IsBinaryType(parameterType.FullName))
+            {
+                bindingDict["DataType"] = "Binary";
+            }
+
             foreach (var property in attribute.GetAllDefinedProperties())
             {
                 bindingDict.Add(property.Key, property.Value);
             }
 
+            // Determine if we should set the "Cardinality" property based on
+            // the presence of "isBatched." This is a property that is from the
+            // attributes that implement the ISupportCardinality interface.
+            //
+            // Note that we are directly looking for "IsBatched" today while we 
+            // are not actually instantiating the Attribute type and instead relying
+            // on type inspection via Mono.Cecil.
+            // TODO: Do not hard-code "isBatched" as the property to set cardinality.
+            // We should rely on the interface
+            // 
+            // Conversion rule
+            //     "isbatched": true => "Cardinality": "Many"
+            //     "isbatched": false => "Cardinality": "One"
+            if (bindingDict.TryGetValue(Constants.IsBatchedKey, out object isBatchedValue)
+                && isBatchedValue is bool isBatched)
+            {
+                // Batching set to true
+                if (isBatched)
+                {
+                    bindingDict["Cardinality"] = "Many";
+                    // Throw if parameter type is *definitely* not a collection type.
+                    // Note that this logic doesn't dictate what we can/can't do, and
+                    // we can be more restrictive in the future because today some 
+                    // scenarios result in runtime failures.
+                    if (IsIterableCollection(parameterType, out DataType dataType))
```

Comment:
Note that this should be changed, but likely for a future iteration it will be a runtime failure anyways, so no worries about introducing breaking behavior (it'll just fail faster)

Reply:
Actually - something that is derived from `IEnumerable<Poco>` (ex: `List<Poco>`) is valid for cardinality: many and **does** have an appropriate converter. Would not want to throw in these cases.

The only failure cases are when element type is `string`, `byte[]`, `ReadOnlyMemory<byte>` `int`, `double`, `long` AND collection type is not equal to IEnumerable, IList, or ICollection (or corresponding generic).

We could go one step further and say if not derived from IEnumerable assume it's not batched... in this case we want to know if it was set explicitly or if it's just from the user... for event hubs we definitely want to discourage non-batched anyways